lib/fastlane/plugin/firebase_app_distribution/actions/firebase_app_distribution_action.rb

```diff
@@ -191,6 +166,47 @@ def self.is_firebasecmd_supported?(cmd)
 
         true
       end
+
+      def self.get_token
+        client = Signet::OAuth2::Client.new(
+          token_credential_uri: 'https://oauth2.googleapis.com/token',
+          client_id: FirebaseAppDistributionLoginAction::CLIENT_ID,
+          client_secret: FirebaseAppDistributionLoginAction::CLIENT_SECRET,
+          refresh_token: ENV["FIREBASE_TOKEN"]
+        )
+
+        client.fetch_access_token!
+        return client.access_token
+      rescue Signet::AuthorizationError => error
+        UI.crash!("Wrong value for FIREBASE_TOKEN")
+      end
+
+      def self.get_app(params)
+        token = get_token
+
+        # begin
+        appId = (params[:app]).to_s
+
+        connection = Faraday.new(url: BASE_URL) do |conn|
+          conn.response(:json, parser_options: { symbolize_names: true })
+          conn.response(:raise_error) # raise_error middleware will run before the json middleware
+          conn.adapter(Faraday.default_adapter)
+        end
+
+        response = connection.get("#{PATH}#{appId}") do |request|
```

Comment:
We only want to catch errors thrown by the `GET` request, so we should surround it in a begin clause and only rescue from errors thrown by the `connection.get` method i.e.

```
begin
  response = connection.get("#{PATH}#{appId}") do |request|
  request.headers["Authorization"] = "Bearer " + token
rescue Faraday::ResourceNotFound => error
  UI.user_error!("App Distribution could not find your app #{params[:app]}. Make sure to onboard your app by pressing the \"Get started\" button on the App Distribution page in the Firebase console: https://console.firebase.google.com/project/_/appdistribution")
rescue => error
  UI.crash!("Failed to fetch app information: #{error.message}")
end

contact_email = response.body[:contactEmail]
if contact_email.strip.empty?
  UI.user_error("We could not find a contact email for app #{appId}. Please visit App Distribution within the Firebase Console to set one up.")
end
```